### PR TITLE
Adding Approximate PCA tests.

### PR DESCRIPTION
**Pull request by etrain**
I've added a few tests to the pca-nodes branch.

I think these should be enough to capture that
1. The sketch algorithm used by the approximate PCA is correct.
2. The net effect we want to capture here (diagonal covariance on dimensionality reduced data) is preserved.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -8,6 +8,7 @@ lib_managed/
 src_managed/
 project/boot/
 project/plugins/project/
+build/
```
Comment (tomerk) [inline .gitignore]: Hmm, why did you need to add this?

**Comment by etrain** [comment]
Updated.

**Comment by tomerk** [comment]
Awesome, LGTM!
